airflow-core/src/airflow/utils/db_cleanup.py

```diff
@@ -491,7 +547,9 @@ def run_cleanup(
     if not dry_run and confirm:
         _confirm_delete(date=clean_before_timestamp, tables=sorted(effective_table_names))
     existing_tables = reflect_tables(tables=None, session=session).tables
-    for table_name, table_config in effective_config_dict.items():
+
+    for table_name in effective_table_names:
+        table_config = effective_config_dict[table_name]
```

Comment:
```suggestion
    for table_name, table_config in effective_config_dict.items():
```

My 2c, the original is better.

Reply:
Pushed a change that keeps the right order - that's why it got changed in the first place.